### DISCUSSION TITLE
How to define split tunneling server side only (using swanctl.conf)?

### DISCUSSION THREAD

**Discussion by acrolink** [strongswan/strongswan#628] [Q&A] [answered]
I see in the Android Client that this can be done by specifying address and subnets inside `custom subnets` field (under Split tunneling).

How to do configure the same server-side only (leaving that field blank in the Android client)? Is the equivalent setting the `child.name.local_ts` or the `child.name.remote_ts` setting, or is it another configuration option?

**Comment by tobiasbrunner** [comment, marked as answer]
Yes, `local_ts` is the correct setting. The Android client will propose `0.0.0.0/0`, which the server narrows to the configured value. If you use clients other than the Android app, have a look at [ForwardingAndSplitTunneling](https://wiki.strongswan.org/projects/strongswan/wiki/ForwardingAndSplitTunneling#Split-Tunneling).

**Reply by acrolink** [reply]
Pondering when the `remote_ts` setting is useful. I read this about in documentation `remote_ts`: 

`Comma separated list of remote selectors to include in CHILD_SA. See local_ts for a description of the selector syntax.`

But that doesn't say much. Is it the server side setting (as opposed to `local_ts` which refers to client side)? Not sure.

**Reply by tobiasbrunner** [reply]
As the documentation says, it refers to the remote traffic selectors. From the server's point of view, it's the source addresses, ports and protocols of traffic from the client (or destination addresses etc. for traffic to the client) that should get tunneled. In roadwarrior scenarios with virtual IPs this is usually not configured (the default is `dynamic` and will be replaced with the assigned virtual IP by the server, see [VirtualIP](https://wiki.strongswan.org/projects/strongswan/wiki/VirtualIP) for details). But a roadwarrior client could e.g. use it to do split-tunneling if the server's `local_ts` is `0.0.0.0/0` (i.e. everything would otherwise get tunneled to the server), as traffic selector narrowing means that the smallest subset of configured and received traffic selectors is selected. For site-to-site scenarios, the traffic selectors allow negotiating the actual subnets etc. that are reachable through the tunnel.
